examples/agents_with_human_in_the_loop.rs

```diff
@@ -0,0 +1,105 @@
+//! This is an example of using a human in the loop pattern with switfide agents.
+//!
+//! In the example we send the tool call over an channel, and then manually approve it.
+//!
+//! In a more realistic example, you can use other rust primitives to make it work for your
+//! usecase. I.e., make an api request with a callback url that will add the feedback.
+//!
+//! Both requesting feedback and providing feedback support an optional payload (as a `serde_json::Value`).
+//!
+//! This allows for more custom workflows, to either display or provide more input to the
+//! underlying tool call.
+//!
+//! For an example on how to implement your own custom wrappers, refer to
+//! `tools::control::ApprovalRequired`
+
+use anyhow::Result;
+use swiftide::{
+    agents::{self, StopReason, tools::control::ApprovalRequired},
+    chat_completion::{ToolCall, ToolOutput, errors::ToolError},
+    traits::AgentContext,
+};
+
+// The macro supports strings/strs, vectors/slices, booleans and numbers.
+//
+// This is currently only supported for the attribute macro, not the derive macro.
+//
+// If you need more control or need to use full objects, we recommend to implement the `Tool` trait
+// and prove the Json spec yourself. Builders are available.
```

Comment:
docs: Add that if you implement your own workflow, there is no need to deal with the hook. Additionally, the tool call is also in the stop reason, the hook is just another example.